go/pkg/cas/upload.go

```diff
@@ -64,6 +64,19 @@ type UploadInput struct {
 	//
 	// This field has no effect if Path is empty.
 	PathExclude *regexp.Regexp
+
+	// OnDigest is called when digest for this item is computed.
+	//
+	// The item is not yet uploaded when OnDigest is called.
+	// It is guaranteed to be fully uploaded only after Client.Upload returns
+	// successfully.
+	//
+	// OnDigest is not guaranteed to be called, e.g.
+	//  - the item is skipped via PathExclude or UploadOptions.Prelude
+	//  - the file at Path is a dangling symlink.
+	//
+	// OnDigest functions are called concurrently.
+	OnDigest func(digest.Digest)
```

Comment:
better to make it possible to return error from this function?

Reply:
done